include/configs/pistachio_bub.h

```diff
@@ -307,18 +304,25 @@
 
 #endif
 
+/*
+ * Openwrt have squashfs rootfs but legacy and other variant uses ubifs
+ * rootfs. To make all compatible dont pass root= for kernel booting
+ * from kernel ubi volume. For others, pass root= to correctly identify
+ * rootfs filesystem.
+ */
 #define NAND_BOOTCOMMAND	\
-	"setenv nandroot ubi.mtd=firmware$boot_partition $nandroot;"\
-	"setenv bootargs $console $earlycon $nandroot $bootextra $mtdparts panic=2;"\
 	"echo Attempting to boot from firmware$boot_partition;"\
+	"setenv ubimtd ubi.mtd=firmware$boot_partition;"\
 	"ubi part firmware$boot_partition || reset;"\
 	"if ubi check kernel; then "\
 		"echo Loading kernel from volume...;"\
+		"setenv bootargs $console $earlycon $ubimtd $bootextra $mtdparts panic=2;"\
 		"ubi read $loadaddr kernel || reset;"\
 	"else "\
 		"echo Loading kernel from rootfs...;"\
+		"setenv bootargs $console $earlycon $ubimtd $nandroot $bootextra $mtdparts panic=2; && "\
```

Comment:
if i dont pass && somehow bootargs was not getting pass correctly to kernel, we could remove ; though. any idea?